File: java/Square.java
public class Square{

    private boolean isHit;
    private String symbol;

    public Square(){
        this.isHit = false;
        this.symbol = "*";
    }

    public void shoot() {
        this.isHit = true;
    }

    public boolean isHit() {
        return isHit;

    }

    public void setSymbol (String symbol) {
        this.symbol = symbol;
    }

    public String toString() {
        return symbol;
    }
}

File: csharp/Square.cs

namespace battleship_warmup_csharp
{
    public class Square
    {
        public bool IsHit { get; private set; }
        public string Symbol { get; set; }

        public Square()
        {
            IsHit = false;
            Symbol = "*";
        }

        public void Shoot()
        {
            IsHit = true;
        }

        public override string ToString() {
            return Symbol;
        }
    }
}